MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs

```diff
@@ -2671,53 +2764,159 @@ public string GetHardwareAcceleratedVideoDecoder(VideoType videoType, MediaStrea
+                                }
+
+                                return null;
+                            }
+
+                            return GetHwaccelType(state, encodingOptions, "hevc");
+                        case "mpeg2video":
+                            return GetHwaccelType(state, encodingOptions, "mpeg2video");
+                        case "vc1":
+                            return GetHwaccelType(state, encodingOptions, "vc1");
+                        case "vp8":
+                            return GetHwaccelType(state, encodingOptions, "vp8");
+                        case "vp9":
+                            if (isColorDepth10)
+                            {
+                                if (encodingOptions.EnableDecodingColorDepth10)
+                                {
+                                    return GetHwaccelType(state, encodingOptions, "vp9");
+                                }
+
+                                return null;
+                            }
+
+                            return GetHwaccelType(state, encodingOptions, "vp9");
                     }
                 }
             }
 
+            var whichCodec = videoStream.Codec.ToLowerInvariant();
+            switch (whichCodec)
+            {
+                case "avc":
+                    whichCodec = "h264";
+                    break;
+                case "h265":
+                    whichCodec = "hevc";
+                    break;
+            }
+
             // Avoid a second attempt if no hardware acceleration is being used
-            encodingOptions.HardwareDecodingCodecs = Array.Empty<string>();
+            encodingOptions.HardwareDecodingCodecs = encodingOptions.HardwareDecodingCodecs.Where(val => val != whichCodec).ToArray();
 
             // leave blank so ffmpeg will decide
             return null;
         }
 
+        /// <summary>
+        /// Gets a hwaccel type to use as a hardware decoder(dxva/vaapi) depending on the system
+        /// </summary>
+        public string GetHwaccelType(EncodingJobInfo state, EncodingOptions options, string videoCodec)
+        {
+            var IsWindows = Environment.OSVersion.Platform == PlatformID.Win32NT;
+            var IsNewWindows = Environment.OSVersion.Version.Major > 6 || (Environment.OSVersion.Version.Major == 6 && Environment.OSVersion.Version.Minor > 1);
+            var IsDxvaSupported = _mediaEncoder.SupportsHwaccel("dxva2") || _mediaEncoder.SupportsHwaccel("d3d11va");
+
+            if ((IsDxvaSupported || IsVaapiSupported(state)) && options.HardwareDecodingCodecs.Contains(videoCodec, StringComparer.OrdinalIgnoreCase))
+            {
+                if (!IsWindows)
+                {
+                    return "-hwaccel vaapi";
+                }
+                else if (IsWindows && IsNewWindows)
+                {
+                    return "-hwaccel d3d11va";
+                }
+                else if (IsWindows && !IsNewWindows)
```

Comment:
```suggestion
                else
```
there's not much choice left anyway :) it's the only possible option left